app/src/lib/stores/app-store.ts

```diff
@@ -1751,6 +1753,35 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this._updateCurrentPullRequest(repository)
     this.updateMenuItemLabels(repository)
     this._initializeCompare(repository)
+    this.repositories = await this.getRefreshedRepositories(
+      this.repositories,
+      repository
+    )
+  }
+
+  public async refreshAllRepositories() {
+    this.repositories = await this.getRefreshedRepositories(this.repositories)
+    this.emitUpdate()
+  }
+
+  private async getRefreshedRepositories(
+    repositories: ReadonlyArray<Repository>,
+    targetRepo?: Repository
+  ): Promise<ReadonlyArray<Repository>> {
+    for (const repo of repositories) {
+      if (!targetRepo || targetRepo.id === repo.id) {
+        await this.withAuthenticatingUser(repo, async (repo, account) => {
+          const gitStore = this.getGitStore(repo)
+          repo.setAheadBehind(gitStore.aheadBehind)
+          await gitStore.fetch(account, true)
+          const status = await gitStore.loadStatus()
+          if (status) {
```

Comment:
We try to be explicit with `null` and `undefined` checks for readability.

`status !== null`